src/lib.rs

```diff
@@ -178,3 +180,49 @@ impl Write for Uart {
 
 // SAFETY: `Uart` just contains a pointer to device memory, which can be accessed from any context.
 unsafe impl Send for Uart {}
+
+impl ErrorType for Uart {
+    type Error = Infallible;
+}
+
+impl Write for Uart {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, Self::Error> {
+        if buf.is_empty() {
+            Ok(0)
+        } else {
+            self.write_byte(buf[0]);
+            Ok(1)
+        }
+    }
+
+    fn flush(&mut self) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+impl WriteReady for Uart {
```

Comment:
The point of the `WriteReady` trait is to allow non-blocking writing. `Write::write` will block until at least one byte can be written, but if the caller wants to avoid blocking they can first call `WriteReady::write_ready` to check whether the UART is ready to write a byte without blocking.

Reply:
Makes sense, thanks!